apps/src/javalab/JavalabEditor.jsx

```diff
@@ -69,37 +81,47 @@ class JavalabEditor extends React.Component {
     };
   };
 
+  renameFileComplete = e => {
+    e.preventDefault();
+    this.props.renameFile(this.state.oldFilename, this.props.filename);
+    this.setState({renameFileActive: false});
+  };
+
+  enableRenameFile() {
+    this.setState({oldFilename: this.props.filename, renameFileActive: true});
+  }
+
   displayFileRename() {
     return (
       <div style={style.tabs}>
-        <div style={style.tab}>
+        <form style={style.renameForm} onSubmit={this.renameFileComplete}>
+          <div style={style.tab}>
+            <input
+              type="text"
+              value={this.props.filename}
+              onChange={e => {
+                this.props.setFilename(e.target.value);
+              }}
+            />
+          </div>
           <input
-            type="text"
-            ariaLabel="file name"
-            value={this.props.fileName}
-            onChange={e => this.props.setFileName(e.target.value)}
-            onBlur={() => this.setState({renameFileActive: false})}
+            className="btn btn-default btn-sm"
+            style={style.button}
+            type="submit"
+            value="Save"
           />
-        </div>
-        <button
-          type="button"
-          onClick={() => this.setState({renameFileActive: false})}
-          className="btn btn-default btn-sm"
-          style={style.button}
-        >
-          Save
-        </button>
+        </form>
       </div>
     );
   }
 
   displayFileNameAndRenameButton() {
     return (
       <div style={style.tabs}>
-        <div style={style.tab}>{this.props.fileName}</div>
+        <div style={style.tab}>{this.props.filename}</div>
         <button
           type="button"
-          onClick={() => this.setState({renameFileActive: true})}
+          onClick={() => this.enableRenameFile()}
```

Comment:
if `enableRenameFile` were an arrow function, this could simplify to `onClick={this.enableRenameFile}`

Reply:
Since this component has a constructor, I'd like to put a plug in for binding it in the constructor.

Reasoning: This is kinda obscure, but our [style guide](https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#react) references the [airbnb style guide](https://github.com/airbnb/javascript/tree/master/react#methods) which has some compelling reasons for binding event handlers exactly once (i.e. it improves performance)
https://github.com/airbnb/javascript/tree/master/react#methods 

TL;DR it would look like this:
```
constructor(props) {
  super(props);
  this.enableRenameFile = this.enableRenameFile.bind(this);
  ...
}

enableRenameFile() {
  ...
}

...

onClick={this.enableRenameFile}
```

Reply:
Similar for the other event handlers.